tests/quickinspectorpickingtest.cpp

```diff
@@ -0,0 +1,208 @@
+    }
+
+private slots:
+    void initTestCase()
+    {
+        qRegisterMetaType<QItemSelection>();
+    }
+
+    void init()
+    {
+        createProbe();
+
+        // we need one view for the plugin to activate, otherwise the model will not be available
+        view = new QQuickView;
+        view->setResizeMode(QQuickView::SizeViewToRootObject);
+        QTest::qWait(1); // event loop re-entry
+
+        itemModel = ObjectBroker::model(QStringLiteral("com.kdab.GammaRay.QuickItemModel"));
+        QVERIFY(itemModel);
+        new ModelTest(itemModel, view);
+
+        sgModel = ObjectBroker::model(QStringLiteral("com.kdab.GammaRay.QuickSceneGraphModel"));
+        QVERIFY(sgModel);
+        new ModelTest(sgModel, view);
+
+        inspector = ObjectBroker::object<QuickInspectorInterface *>();
+        QVERIFY(inspector);
+        inspector->selectWindow(0);
+        QTest::qWait(1);
+    }
+
+    void cleanup()
+    {
+        delete view;
+        QTest::qWait(1);
+    }
+
+    void testItemPicking_data()
+    {
+        QTest::addColumn<QString>("qmlFile", nullptr);
+        QTest::addColumn<QString>("pickedObjectId", nullptr);
+
+        QTest::newRow("Vanilla Rect-Clicking") << "qrc:/manual/picking/stackedrects.qml" << "bluerect";
+        QTest::newRow("Negative z-order") << "qrc:/manual/picking/negativezordering.qml" << "greenrect";
+        QTest::newRow("Invisible overlay") << "qrc:/manual/picking/invisibleoverlay.qml" << "redrect";
+        QTest::newRow("Opacity:0 overlay") << "qrc:/manual/picking/opacityzerooverlay.qml" << "yellowrect";
+        QTest::newRow("Loader") << "qrc:/manual/picking/loader.qml" << "bluerect";
+        //Does not Work
+        QTest::newRow("Outside of parent") << "qrc:/manual/picking/outsideofparent.qml" << "redrectchild";
+    }
+
+    // Info: Clickposition is always in Center of View
+    void testItemPicking()
+    {
+        QFETCH(QString, qmlFile);
+        QFETCH(QString, pickedObjectId);
+
+        QVERIFY(showSource(qmlFile));
+
+        ToolManagerInterface *toolManager = ObjectBroker::object<ToolManagerInterface *>();
+        QSignalSpy toolSpy(toolManager, SIGNAL(toolSelected(QString)));
+        QVERIFY(toolSpy.isValid());
+
+        auto itemSelectionModel = ObjectBroker::selectionModel(itemModel);
+        QVERIFY(itemSelectionModel);
+        QSignalSpy itemSpy(itemSelectionModel, SIGNAL(selectionChanged(QItemSelection,QItemSelection)));
+        QVERIFY(itemSpy.isValid());
+
+
+        auto sgSelectionModel = ObjectBroker::selectionModel(sgModel);
+        QVERIFY(sgModel);
+        QSignalSpy sgSpy(sgSelectionModel, SIGNAL(selectionChanged(QItemSelection,QItemSelection)));
+        QVERIFY(sgSpy.isValid());
+
+        // auto center-click is broken before https://codereview.qt-project.org/141085/
+        QTest::mouseClick(view, Qt::LeftButton, Qt::ShiftModifier | Qt::ControlModifier,
+                          QPoint(view->width()/2, view->height()/2));
+        QTest::qWait(20);
```

Comment:
hardcoded timing are brittle on the CI, maybe better use QVERIFY(xxxSpy.wait())?